### DISCUSSION TITLE
Changing section titles, how to edit "Sponsor Me!" in the Sponsors plugin?

### DISCUSSION THREAD

**Discussion by lissy93** [lowlighter/metrics#1126] [Ideas]
Is there a way to change the default title of a plugin section? 

Specifically, in the Sponsor plugin, I would really like the title to be something like _**"Thank you"**_ instead of **_"Sponsor Me"_**.

[![example of sponsors plugin](https://raw.githubusercontent.com/Lissy93/Lissy93/master/assets/metrics/sponsors.svg)](https://github.com/Lissy93/Lissy93)

---

I'd be happy to try and submit a PR, if this is something that might be useful to others? I'm not sure if this is just a case of replacing the [hard-coded value](https://github.com/lowlighter/metrics/blob/master/source/templates/classic/partials/sponsors.ejs#L5) with something like `plugins.sponsors.title || "Sponsor me!"`, and adding it the action.yml file and the docs. Or is there's more to it than that?

---

I also wanted to say a big thank you to @lowlighter - This project is super awesome, and everything is beautifully documented  🙌
I've had a lot of fun playing around with it, and it's clear how much time you've put towards building it. So I've sent you a couple of quid so you can get yourself a few drinks, as a small thank you.

**Comment by lowlighter** [comment]
Hi 👋 !

Thanks a lot for your kind words and support, I really appreciate it and am happy you like this project 😄 !

I think indeed a `plugin_sponsors_title` would make a lot of sense in this context.
The website `screenshot` plugin actually have a similar feature where you can change the section title:
https://github.com/lowlighter/metrics/blob/1713f5e47f319f33a91f24e8927b39decc59e3dd/source/plugins/community/screenshot/metadata.yml#L26-L30
___

I'd be happy to see such a feature, so PRs are welcome 😄 

To make this project easier to maintain, `action.yml` and docs are mostly auto-generated, so they shouldn't be edited directly.

You'll need instead to define the `plugin_sponsors_title` option within the [`metadata.yml`](https://github.com/lowlighter/metrics/blob/master/source/plugins/sponsors/metadata.yml) (which serves as specification, documentation and input validation) of the sponsors plugin. 

When `imports.metadata.plugins.sponsors.inputs()` is called, it'll give you a compliant `title` value to what was specified in the `metadata.yml` that you can extract, similarly to the other inputs:
https://github.com/lowlighter/metrics/blob/1713f5e47f319f33a91f24e8927b39decc59e3dd/source/plugins/sponsors/index.mjs#L10

Then just send it back at the end to expose it in templating:
https://github.com/lowlighter/metrics/blob/1713f5e47f319f33a91f24e8927b39decc59e3dd/source/plugins/sponsors/index.mjs#L84

And you'll be able to patch the template by using `<%= plugins.sponsors.title %>` instead like you suggested:
https://github.com/lowlighter/metrics/blob/1713f5e47f319f33a91f24e8927b39decc59e3dd/source/templates/classic/partials/sponsors.ejs#L5

It's not necessary to add a test case, since they only check that code is able to run, they aren't capable of checking the content of the resulting SVG atm

____

Hope this helps 👍 !

**Reply by lissy93** [reply]
Awesome, thank you for the detailed explanation. I've submitted a PR in #1128 :)
